sips/sip-027/sip-027-non-sequential-multisig-transactions.md

```diff
@@ -0,0 +1,413 @@
+   ECDSA recoverable signature) pair to recover the public key that generated it.
+
+For sequential hash modes `0x00`, `0x01`, `0x02`, and `0x03`:
+
+5. Calculate the `postsign-sighash` from `presign-sighash`, the signature, public key encoding
+   byte,
+6. Repeat steps 3-5 for each signature, so that all of the public keys are
+   recovered.
+7. Verify that the sequence of public keys hash to the address, using
+   the address's indicated public key hashing algorithm, and the number of signatures
+   matches **exactly** the required number of signatures.
+
+For non-sequential hash modes `0x05` and `0x07`:
+
+5. Repeat step 4 for each signature, so that all of the public keys are
+   recovered.
+6. Verify that the sequence of public keys hash to the address, using
+   the address's indicated public key hashing algorithm, and the number of signatures
+   is **at least** the required number of signatures.
+
+When verifying a sponsored transaction, the sponsor's signatures are verified
+first.  Once verified, the sponsor spending condition is set to the "signing
+sentinel" value in order to verify the origin spending condition.
+
+#### Additional Recommendations
+
+While this SIP allows signers to sign in any order, the ordering of public keys in the transaction auth fields still affects multisig account address generation.
+When funding a multisig account or creating a transaction, it is strongly recommended, but not required, to order public keys from least to greatest (equivalant to lexographically sorting the hex-encoded strings).
+This will remove the requirement to remember key order and result in consistent address generation.
+
+# Related Work
+
+[PR #139](https://github.com/stacksgov/sips/pull/139): This draft SIP was created earlier but lacked the technical specifications for implementation. The author has since closed this PR in favor of this draft
+
+# Layer
+
+Consensus (hard fork)
+
+# Requires
+
+[SIP-005](https://github.com/stacksgov/sips/blob/main/sips/sip-005/sip-005-blocks-and-transactions.md)
+
+# Backwards Compatibility
+
+The Stacks Blockchain will continue to treat multisig transactions using the current format as valid.
+Existing multisig accounts will be able to use the new transaction types to spend previously recieved funds.
+
+# Activation
+
+Since this SIP requires a change to the stacks consensus rules a community vote is additionally required.
+
+## Process of Activation
+Users can vote to approve this SIP with either their locked/stacked STX or with unlocked/liquid STX, or both. The criteria for the stacker and non-stacker voting is as follows.
+
+## For Stackers:
+
+In order for this SIP to activate, the following criteria must be met by the set of Stacked STX:
+
+- At least double the amount of Stacked STX locked by the largest Stacker in the cycle preceding the vote must vote at all to activate this SIP.
+- Of the Stacked STX that vote, at least 70% of them must vote "yes."
```

Comment:
Just looking at the SIP-27 while reviewing SIP-28, noticed slight discrepancy 
between the [public blog announcement](https://stacks.org/sip-027-voting) of the Stackers' voting threshold vs the SIP text. 

Recommend update the SIP text to match the blog. 

i.e. 

* At least 80 million Stacked STX must vote at all to activate this SIP.

* Of the Stacked STX that vote, at least 80% of them must vote "yes."

Reply:
> Just looking at the SIP-27 while reviewing SIP-28, noticed slight discrepancy 
> between the [public blog announcement](https://stacks.org/sip-027-voting) of the Stackers' voting threshold vs the SIP text. 
> 
> Recommend update the SIP text to match the blog. 
> 
> i.e. 
> 
> * At least 80 million Stacked STX must vote at all to activate this SIP.
> 
> * Of the Stacked STX that vote, at least 80% of them must vote "yes."
>